packages/modules/psa/api.py

```diff
@@ -0,0 +1,130 @@
+        client_id = manufacturer_config.client_id
+        client_secret = manufacturer_config.client_secret
+
+    data = {'realm': realm, 'grant_type': 'password', 'password': password, 'username': user_id, 'scope': scope}
+    psa_config = req.get_http_session().post(
+        "https://idpcvs." + str(brand) + "/am/oauth2/access_token", data=data, auth=(client_id, client_secret)
+        ).json()
+    access_token = psa_config['access_token']
+
+    # Step 2: Retrieve VINs and VIN_IDs in profile
+    data = {'client_id': client_id}
+    headers = {
+        'Accept': 'application/hal+json', 'Authorization': 'Bearer %s' % access_token, 'x-introspect-realm': realm}
+
+    vehicle_response = req.get_http_session().get(
+        "https://api.groupe-psa.com/connectedcar/v4/user/vehicles", params={"client_id": client_id}, headers=headers
+        ).json()
+    vehicles = vehicle_response['_embedded']['vehicles']
+
+    # Filter list for given VIN or select first entry if no VIN is provided
+    try:
+        vehicle_selected = next(
+            vehicle for vehicle in vehicles if vehicle['vin'] == vehicle_vin) if vehicle_vin else vehicles[0]
+        vehicle_id = vehicle_selected['id']
+        vehicle_vin_found = vehicle_selected['vin']
+    except Exception:
+        log.error("Cannot find VIN '%s' in PSA account %s. Check VIN and retry.'", vehicle_vin, user_id)
+        return
+
+    log.debug("Fetching details for VIN: %s with vehicle_id:%s", vehicle_vin_found, vehicle_id)
+
+    # Step 3: get details for selected vehicle
+    data = {'client_id': client_id}
+    headers = {
+        'Accept': 'application/hal+json', 'Authorization': 'Bearer %s' % access_token, 'x-introspect-realm': realm}
+    battery = req.get_http_session().get(
+        "https://api.groupe-psa.com/connectedcar/v4/user/vehicles/" + str(vehicle_id) + "/status",
+        params={"client_id": client_id}, headers=headers).json()
+
+    # filter to only include type=Electric but remove all others. Seen type=Fuel and type=Electric being returned.
+    energies = battery['energy']
+    energy_selected = next(energy for energy in energies if energy['type'] == 'Electric')
+    soc = energy_selected['level']
+    range = energy_selected['autonomy']
+    soc_timestamp = energy_selected['updatedAt']
+
+    return soc, range, soc_timestamp
+
+
+def fetch_soc(user_id: str,
+              password: str,
+              client_id: str,
+              client_secret: str,
+              manufacturer: str,
+              soccalc: bool,
+              vehicle_vin: str,
+              vehicle: int) -> tuple:
+
+    try:
+        soc, range, soc_timestamp = requestPSAData(
+            user_id, password, client_id, client_secret, manufacturer, soccalc, vehicle_vin, vehicle)
+    except Exception as err:
+        log.error("psa.fetch_soc: requestData Error, vehicle: %s error: %s, %s", str(vehicle), err, type(err))
+        raise
```

Comment:
Ich vermute deine Motiviation ist der Exception eine weitere Information hinzuzufügen - den Inhalt der Variable "vehicle". Das kannst du machen, besser ist aber der bestehenden Exception weitere Informationen hinzuzufügen indem du noch eine Exception wirfst. Also:
```suggestion
        raise Exception("Error requesting PSA data for vehicle: %s" % vehicle)
```
Die alte Exception ist immernoch da und wird zusammen mit der Meldung von hier auch geloggt (alles organisiert vom `SingleComponentUpdateContext`).

Tatsächlich kennt der `SingleComponentUpdateContext` den Inhalt von `vehicle` sowieso, den die Variable steht in der `ComponentInfo`. Daher würde ich sagen, du kannst den `except`-Block hier gefahrlos ersatzlos streichen.